iOS/Flows/Conversation/Conversation List/ConversationListViewController+Extensions.swift

```diff
@@ -84,19 +84,19 @@ extension ConversationListViewController {
         ConversationsManager.shared.$activeConversation.mainSink { conversation in
             if let convo = conversation,
                let cell = self.collectionView.getCentermostVisibleCell() as? ConversationMessagesCell {
-                self.handleTopMessageUpdates(for: convo, cell: cell)
+                self.subscribeToTopMessageUpdates(for: convo, cell: cell)
             }
         }.store(in: &self.cancellables)
     }
     
-    func handleTopMessageUpdates(for conversation: Conversation, cell: ConversationMessagesCell) {
-    
-        /// didUpdate is called before this is ever set. Also looks like a non centered conversation is being used
-        cell.$incomingTopmostMessage
+    func subscribeToTopMessageUpdates(for conversation: Conversation, cell: ConversationMessagesCell) {
+        // didUpdate is called before this is ever set.
+        // Also looks like a non centered conversation is being used
+        self.topMessageSubscription = cell.$incomingTopmostMessage
+            .removeDuplicates()
             .mainSink { [unowned self] message in
                 guard let author = message?.author else { return }
                 self.headerVC.membersVC.updateAuthor(for: conversation, user: author)
-                
-            }.store(in: &self.cancellables)
```

Comment:
This subscription was never getting cancelled, so the cell could get subscribed to multiple times which was created memory leaks and wasted computation.